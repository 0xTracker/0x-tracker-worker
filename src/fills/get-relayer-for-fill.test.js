const getRelayerForFill = require('./get-relayer-for-fill');

it('should return null when no relayer matched', () => {
  const fill = {
    feeRecipient: '0xe269e891a2ec8585a378882ffa531141205e92e9',
    takerAddress: '0x5dd835a893734b8d556eccf87800b76dda5aedc5',
  };
  const relayer = getRelayerForFill(fill);

  expect(relayer).toBeNull();
});

it('should return DDEX when taker address matches', () => {
  const fill = { takerAddress: '0xe269e891a2ec8585a378882ffa531141205e92e9' };
  const relayer = getRelayerForFill(fill);

  expect(relayer).toMatchSnapshot();
});

it('should return Bamboo Relay when fee recipient matches', () => {
  const fill = { feeRecipient: '0x5dd835a893734b8d556eccf87800b76dda5aedc5' };
  const relayer = getRelayerForFill(fill);

  expect(relayer).toMatchSnapshot();
});

it('should return Gods Unchained when sender address matches', () => {
  const fill = { senderAddress: '0xb04239b53806ab31141e6cd47c63fb3480cac908' };
  const relayer = getRelayerForFill(fill);

  expect(relayer.name).toBe('Gods Unchained');
});
