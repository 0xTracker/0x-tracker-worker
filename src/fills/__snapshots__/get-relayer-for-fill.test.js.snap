// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return Bamboo Relay when fee recipient matches 1`] = `
Object {
  "feeRecipients": Array [
    "0x5dd835a893734b8d556eccf87800b76dda5aedc5",
  ],
  "id": "bambooRelay",
  "imageUrl": "https://0xtracker.com/assets/logos/bamboo-relay.png",
  "lookupId": 1,
  "name": "Bamboo Relay",
  "slug": "bamboo-relay",
  "url": "https://bamboorelay.com/",
}
`;

exports[`should return DDEX when taker address matches 1`] = `
Object {
  "id": "ddex",
  "imageUrl": "https://0xtracker.com/assets/logos/ddex.png",
  "lookupId": 2,
  "name": "DDEX",
  "orderMatcher": true,
  "slug": "ddex",
  "takerAddresses": Array [
    "0xe269e891a2ec8585a378882ffa531141205e92e9",
  ],
  "url": "https://ddex.io",
}
`;
